aws/resource_aws_iam_group_policy.go

```diff
@@ -58,7 +62,13 @@ func resourceAwsIamGroupPolicyPut(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	var policyName string
-	if v, ok := d.GetOk("name"); ok {
+	var err error
+	if !d.IsNewResource() {
```

Comment:
Is this new logic necessary? I think if it is, rather than introducing this new `d.IsNewResource()` logic here, instead we typically prefer to split the resource `Create` and `Update` functions to remove any confusion. The `Update` function can be created by duplicating the _prior_ `resourceAwsIamGroupPolicyPut()` function and using `resourceAwsIamGroupPolicyParseId()`.

Reply:
Actually I guess we can keep it as before. I reverted it.